src/modules/video_player/index.js

```diff
@@ -128,6 +148,44 @@ class VideoPlayerModule {
             hidePlayerCursor();
         });
     }
+
+    volumeClickHandler() {
+        this.isMutedByUser = !this.isMutedByUser;
+    }
+
+    onWindowBlur() {
+        const currentPlayer = twitch.getCurrentPlayer();
+        if (!currentPlayer || !currentPlayer.player || !document.hidden) return;
+        currentPlayer.player.setMuted(true);
+    }
+
+    onWindowFocus() {
+        const currentPlayer = twitch.getCurrentPlayer();
+        if (!currentPlayer || !currentPlayer.player) return;
+        currentPlayer.player.setMuted(this.isMutedByUser);
+    }
+
+    onVisibilityChange() {
+        if (document.hidden) {
+            this.onWindowBlur();
+        } else {
+            this.onWindowFocus();
+        }
+    }
+
+    muteInvisibleTabs() {
+        $(document).off('visibilitychange', this.onVisibilityChange);
+        $(window).off('blur', this.onWindowBlur);
```

Comment:
why do we support both visibility and blur? window could be blurred but still visible, right?

Reply:
The original issue was with the logic of getting the player mute state, but this no longer applies. I've switched to `visibilitychange` only.